### ISSUE TITLE
ComboBox with IsEditable="true" can get unselected when navigating away

### ISSUE THREAD

**Issue by mcooley** [microsoft/microsoft-ui-xaml#5756]
**Describe the bug**
If a user clicks away from an editable combo box without selecting a value, the selected value is cleared.

**Steps to reproduce the bug**
```
<ComboBox
    x:Name="MyComboBox"
    IsEditable="True"
    Loaded="ComboBox_Loaded">
        <x:String>20</x:String>
        <x:String>40</x:String>
        <x:String>60</x:String>
</ComboBox>

private void ComboBox_Loaded(object sender, RoutedEventArgs e)
{
    this.MyComboBox.SelectedItem = "40";
}
```
When the application launches, click the drop down arrow to open the combo box, then click away.
Expected behavior: "40" is still selected
Actual behavior: nothing is selected

This behavior occurs only if the ComboBox has IsEditable="true", and only when the ComboBox is first loaded.

**Version Info**
Windows 22000, occurs with or without WinUI 2 present

**Comment by mcooley** [comment]
@StephenLPeters I see this is marked as "needs-winui-3". Because of the number of issues we're seeing with ComboBox in our "font picker" scenario (see #5733, #5734, #5735), we'd be very interested in seeing a revised version of ComboBox in WinUI 2. Is a WinUI 2 port for this control something that you'd consider, or accept a contribution for?

**Comment by StephenLPeters** [comment]
@mcooley The issue is that this behavior is dictated by the comboBox's code behind, not its visual template.  Sometimes we can fix issues with a simple retemplate that we publish in Winui2 however in this case this issue isn't possible to fix that way.  So we have 3 options.

1. We could fix the issue in the code behind in the OS. The issue here is that the fix is OS bound meaning that it won't actually fix the issue for the majority of users until they update to the new OS version. Additionally there is significant overhead with cracking the OS and thus the team very rarely finds this solution acceptable.
2. We could fix the issue in Winui3.  This is our default for issues like this. Winui3 is in active development and is the teams vision for the future.
3. We could publish a brand new control with the same name as ComboBox. So Controls:ComboBox and ComboBox would be different code behinds.  This is what you are suggesting as a possible fix, however it is a significant ask.  Recreating this control has a huge cost and is not something we can afford right now (@ranjeshj can confirm).  We could discuss accepting a contribution of this, however the initial development cost is not the only expense the team is worried about.

@ranjeshj FYI

**Comment by sharpninja** [comment]
This is still broken, nearly two years later.

**Comment by DmitriyKomin** [comment]
@mcooley, thank your for reporing this. Unfortunately we are not currently investigating WinUI2 issues that are not business or security critical (more on the policy [here](https://github.com/microsoft/microsoft-ui-xaml/blob/main/docs/contribution_handling.md#winui-2)).

We believe this is fixed in WinUI3 - cannot repro on WASDK 1.5 and latest WinUI3 Gallery.
